hubblestack/extmods/hubble_mods/curl.py

```diff
@@ -0,0 +1,238 @@
+            params:
+                key1: val1
+                key2: val2
+            data: "payload for post/put"
+            headers:
+                header1: val1
+                header2: val2
+            username: user
+            password: pwd
+            timeout: 9
+            decode_json: True
+
+          comparator:
+            type: "string"
+            match: "host*"
+            is_regex: true
+
+Note: You can use any comparator depending upon the kind of data a url is returning.
+
+FDG Example:
+------------
+main:
+  description: 'curl check'
+  module: curl
+    args:
+        url: "https://adobe.com/someurl"
+        function: GET
+        params:
+            key1: val1
+            key2: val2
+        data: "payload for post/put"
+        headers:
+            header1: val1
+            header2: val2
+        username: user
+        password: pwd
+        timeout: 9
+        decode_json: True
+
+Mandatory Params:
+    url
+"""
+import os
+import logging
+import requests
+
+import hubblestack.extmods.module_runner.runner_factory as runner_factory
+import hubblestack.extmods.module_runner.runner_utils as runner_utils
+from hubblestack.utils.hubble_error import HubbleCheckValidationError
+from salt.exceptions import CommandExecutionError
+
+log = logging.getLogger(__name__)
+
+
+def validate_params(block_id, block_dict, extra_args=None):
+    """
+    Validate all mandatory params required for this module
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'caller': 'Audit'}
+
+    Raises:
+        AuditCheckValidationError: For any validation error
+    """
+    log.debug('Module: curl Start validating params for check-id: {0}'.format(block_id))
+
+    # fetch required param
+    function_name = runner_utils.get_param_for_module(block_id, block_dict, 'function')
+    if not function_name:
+        function_name = 'GET'
+    
+    url = runner_utils.get_param_for_module(block_id, block_dict, 'url')
+    
+    error = {}
+    if function_name not in ('GET', 'PUT', 'POST'):
+        error['function'] = 'Invalid request type: {0}'.format(function_name)
+
+    if not url:
+        error['url'] = 'URL not passed'
+
+    if error:
+        raise HubbleCheckValidationError(error)
+
+    log.debug('Validation success for check-id: {0}'.format(block_id))
+
+def execute(block_id, block_dict, extra_args=None):
+    """
+    For getting params to log, in non-verbose logging
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+
+    returns:
+        tuple of result(value) and status(boolean)
+    """
+    log.debug('Executing Curl module for id: {0}'.format(block_id))
+
+    chain_args = None if not extra_args else extra_args.get('chaining_args')
+    if chain_args:
```

Comment:
Need to remove this statement now.

Reply:
This is to print warning message. Chained values are not being used anywhere.